### PR TITLE
[T2 chassis] platform and snmp tests that uses pdu_controller to run on sup for chassis

### PR DESCRIPTION
**Pull request by wenyiz2021**
### Description of PR

1.fixture `pdu_controller`: implement a function `get_sup_node_or_first_node`for chassis whose pdu info is only avail on sup. Note that here we don't use `enum_supervisor_dut_hostname`, because `platform_tests/conftest.py` have a autouse fixture `skip_on_simx` that makes all platform tests use `rand_one_dut_hostname`.

2. other platform tests who uses fixture `pdu_controller` also need same function implementation and usage.
3. snmp tests is fine with `enum_supervisor_dut_hostname`

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### How did you verify/test it?
Verified by running all tests changed in this PR on Cisco chassis-3

### CODE REVIEW

**Review by arlakshm — COMMENTED** [review]
With this change this test will be skipped some times and other times it will be executed on sup.
Can update the tests to run on the supervisor always?

**Comment by wenyiz2021** [comment]
> With this change this test will be skipped some times and other times it will be executed on sup. Can update the tests to run on the supervisor always?

Hi @arlakshm , in previous commit I changed to run on the supervisor for `test_turn_on_off_psu_and_check_psustatus`, however, it uses fixture `pdu_controller` which is used in a lot of other tests, and it uses `enum_rand_one_per_hwsku_hostname`. So I have to keep current fixture, do you have any better suggestions?

**Comment by wenyiz2021** [comment]
> > With this change this test will be skipped some times and other times it will be executed on sup. Can update the tests to run on the supervisor always?
> 
> Hi @arlakshm , in previous commit I changed to run on the supervisor for `test_turn_on_off_psu_and_check_psustatus`, however, it uses fixture `pdu_controller` which is used in a lot of other tests, and it uses `enum_rand_one_per_hwsku_hostname`. So I have to keep current fixture, do you have any better suggestions?

oh, I think all tests calling `pdu_controller` need to use SUP then. because `pdu_controller` can only get from SUP..

File: tests/common/plugins/pdu_controller/__init__.py [hunk 1]
```diff
@@ -30,15 +30,28 @@ def get_pdu_visible_vars(inventories, pdu_hostnames):
     return pdu_hosts_vars
 
 
+def get_sup_node_or_first_node(duthosts):
```
Comment (arlakshm) [inline tests/common/plugins/pdu_controller/__init__.py:33]: can we define this function in common file in `utilities.py`?

File: tests/common/plugins/pdu_controller/__init__.py [hunk 2]
```diff
@@ -30,15 +30,28 @@ def get_pdu_visible_vars(inventories, pdu_hostnames):
...
+    # accomodate for T2 chassis, which only SUP has pdu info
+    # single-dut get itself
+    if len(duthosts) == 1:
+        return duthosts[0]
+    # try to find sup node in multi-dut
+    for dut in duthosts:
+        if dut.is_supervisor_node():
+            return dut
+    # if not chassis, it's dualtor, return first node
+    return duthosts[0]
```
Comment (arlakshm) [inline tests/common/plugins/pdu_controller/__init__.py:43]: should be return random host here instead of always returning first dut?

Comment (wenyiz2021) [reply tests/common/plugins/pdu_controller/__init__.py:43]: this is how enum_supervisor_duthost is implemented

File: tests/common/plugins/pdu_controller/__init__.py (hunk 1 above)
Comment (wenyiz2021) [reply tests/common/plugins/pdu_controller/__init__.py:33]: will do, thanks

**Pull request closed without merging**